### PR TITLE
PAYARA-4037 Add access log setting options for Payara Micro

### PR DESCRIPTION
**Pull request by AlanRoth**
# Description

This is a feature for Payara Micro.

Introduces 3 new Micro Commands, 
`accessloginterval` Sets the access log write-interval attribute 
`accesslogsuffix` Sets the access log rotation-suffix attribute 
`accesslogprefix` Sets a new virtual server property that will be used as the prefix

# Testing

### Testing Performed
Build tests

`java -jar payara-micro --accesslog ~/temp/accesslog --accessloginterval 0 --unpackdir ~/temp`
Check in domain.xml in ~/temp/config that `write-interval-seconds` has a value of 0

`java -jar payara-micro --accesslog ~/temp/accesslog --accesslogsuffix ss --unpackdir ~/temp`
Check in domain.xml in ~/temp/config that `rotation-suffix` is set to `ss` (seconds in simple date format)
Check in ~/temp/accesslog that accesslogs are named simmilar to; `server_access_log_32`

`java -jar payara-micro --accesslog ~/temp/accesslog --accesslogprefix testy --unpackdir ~/temp`
Check in ~/temp/accesslog that the file is named `testy2019-11-22` or simmilar exists

`java -jar payara-micro --accesslog ~/temp/accesslog --accesslogprefix testy --accesslogsuffix ss --unpackdir ~/temp`
Check in ~/temp/accesslog that a file named `testy32` or simmilar exists

### Testing Environment
Manjaro 18.0.4
Maven 3.6.1
Zulu JDK 1.8.0_222

# Documentation
Soon to follow

# Notes for Reviewers
Feel free to mix and match the commands to make sure they are compatible, from testing the commands work with all existing commands

### CODE REVIEW

**Comment by AlanRoth** [comment]
jenkins test please

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 1]
```diff
@@ -2373,6 +2404,14 @@ private void packageUberJar() {
             props.setProperty("payaramicro.userLogFile", userLogFile);
         }
 
+        if (accessLogInterval != Integer.MIN_VALUE) {
+            props.setProperty("payaramicro.accessLogInterval", Integer.toString(accessLogInterval));
+        }
+        
+        if (accessLogPrefix != null) {
+            props.setProperty("payaramicro.customlog", accessLogPrefix);
```
Comment (Pandrex247) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: Could do with a better name

Comment (AlanRoth) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]:
```suggestion
            props.setProperty("payaramicro.accessLogPrefix", accessLogPrefix);
```

Comment (Pandrex247) [reply appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]:
```suggestion
            props.setProperty("payaramicro.accessLogPrefix", accessLogPrefix);
```

File: appserver/web/web-glue/src/main/java/com/sun/enterprise/web/PEAccessLogValve.java [hunk 2]
```diff
@@ -766,7 +767,12 @@ boolean updateVirtualServerProperties(
                         acBufferSize);
             }
         }
-
+        
+        String acFilePrefix = vsBean.getPropertyValue(Constants.ACCESS_LOG_PREFIX);
+        if(acFilePrefix != null && !acFilePrefix.trim().isEmpty() && !"null".equals(acFilePrefix) && !"null".equals(accessLogPrefix)) {
```
Comment (AlanRoth) [inline appserver/web/web-glue/src/main/java/com/sun/enterprise/web/PEAccessLogValve.java]: Hacky but does the job, caused some grief and NPE

File: nucleus/payara-modules/hazelcast-bootstrap/src/test/java/fish/payara/nucleus/hazelcast/test/DNSDiscoveryTest.java [hunk 3]
```diff
@@ -72,7 +72,7 @@ public void multipleRecordTest() {
     }
     
     @Test
-    public void sinlgeRecordTest() {
+    public void singleRecordTest() {
```
Comment (AlanRoth) [inline nucleus/payara-modules/hazelcast-bootstrap/src/test/java/fish/payara/nucleus/hazelcast/test/DNSDiscoveryTest.java]: Unrelated, just a typo fix

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java (hunk 1 above)
Comment (Pandrex247) [reply appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: Actually is this even used anywhere? Doesn't seem to be...?

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java [hunk 4]
```diff
@@ -2373,6 +2404,14 @@ private void packageUberJar() {
             props.setProperty("payaramicro.userLogFile", userLogFile);
         }
 
+        if (accessLogInterval != Integer.MIN_VALUE) {
+            props.setProperty("payaramicro.accessLogInterval", Integer.toString(accessLogInterval));
```
Comment (Pandrex247) [inline appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: Neither does this?

File: appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java (hunk 1 above)
Comment (AlanRoth) [reply appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/PayaraMicroImpl.java]: I've spotted that after making the PR, and I added it for consistency with the rest of the properties

File: nucleus/payara-modules/hazelcast-bootstrap/src/test/java/fish/payara/nucleus/hazelcast/test/DNSDiscoveryTest.java (hunk 3 above)
Comment (cubastanley) [reply nucleus/payara-modules/hazelcast-bootstrap/src/test/java/fish/payara/nucleus/hazelcast/test/DNSDiscoveryTest.java]: Was this change made as a codebase wide refactor and not just a renaming of the method in the class? Assuming yes but always worth asking ;)

**Comment by AlanRoth** [comment]
> 
> 
> The suffix and prefix don't actually seem to be suffixes or prefixes - they overwrite stuff.
> 
> For example, original log file: `server_access_log.2019-11-22.txt`
> With your options: `testy07.txt`
> 
> Surely it should be something more like `testy.server_access_log.07.txt`?

technically server_access_log is the default prefix or more specifically `{virtualServer id} _access_log` and 2019-11-22 being the suffix, I'm happy to change it though

File: nucleus/payara-modules/hazelcast-bootstrap/src/test/java/fish/payara/nucleus/hazelcast/test/DNSDiscoveryTest.java (hunk 3 above)
Comment (AlanRoth) [reply nucleus/payara-modules/hazelcast-bootstrap/src/test/java/fish/payara/nucleus/hazelcast/test/DNSDiscoveryTest.java]: No, but yes. I didn't check but luckily thats the only occurrence of the method

**Comment by AlanRoth** [comment]
jenkins test please

**Comment by AlanRoth** [comment]
jenkins test please

**Pull request merged**